rcl/src/rcl/subscription.c

```diff
@@ -193,6 +209,290 @@ rcl_subscription_get_default_options()
   return default_options;
 }
 
+rcl_ret_t
+rcl_subscription_options_fini(rcl_subscription_options_t * option)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(option, RCL_RET_INVALID_ARGUMENT);
+  // fini rmw_subscription_options_t
+  const rcl_allocator_t * allocator = &option->allocator;
+  RCL_CHECK_ALLOCATOR_WITH_MSG(allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
+
+  if (option->rmw_subscription_options.content_filter_options) {
+    rmw_ret_t ret = rmw_subscription_content_filter_options_fini(
+      option->rmw_subscription_options.content_filter_options, allocator);
+    if (RCUTILS_RET_OK != ret) {
+      RCUTILS_SAFE_FWRITE_TO_STDERR("Failed to fini content filter options.\n");
+      return rcl_convert_rmw_ret_to_rcl_ret(ret);
+    }
+    allocator->deallocate(
+      option->rmw_subscription_options.content_filter_options, allocator->state);
+    option->rmw_subscription_options.content_filter_options = NULL;
+  }
+  return RCL_RET_OK;
+}
+
+rcl_ret_t
+rcl_subscription_options_set_content_filter_options(
+  const char * filter_expression,
+  size_t expression_parameters_argc,
+  const char * expression_parameter_argv[],
+  rcl_subscription_options_t * options)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(filter_expression, RCL_RET_INVALID_ARGUMENT);
+  if (expression_parameters_argc > 100) {
+    RCL_SET_ERROR_MSG("The maximum of expression parameters argument number is 100");
+    return RCL_RET_INVALID_ARGUMENT;
+  }
```

Comment:
isn't this check already done in rmw?

Reply:
Considering rmw_connextdds has its [special extension setting](https://github.com/ros2/rmw/pull/302#discussion_r827572015) and a fallback for cft in the `rcl`, so I need to add this checkment in `rcl`.